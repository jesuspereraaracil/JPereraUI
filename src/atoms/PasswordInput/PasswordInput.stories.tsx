import type { Meta, StoryObj } from '@storybook/react';
import { PasswordInput } from './PasswordInput';

const meta: Meta<typeof PasswordInput> = {
  /* ðŸ‘‡ The title prop is optional.
   * See https://storybook.js.org/docs/react/configure/overview#configure-story-loading
   * to learn how to generate automatic titles
   */
  title: 'Atoms/PasswordInput',
  component: PasswordInput,
};
export default meta;
type Story = StoryObj<typeof PasswordInput>;

export const Default: Story = {
  args: {
    error: false,
    disabled: false,
  },
};

export const Error: Story = {
  args: {
    error: true,
    disabled: false,
  },
};

export const Disabled: Story = {
  args: {
    error: false,
    disabled: true,
  },
};
