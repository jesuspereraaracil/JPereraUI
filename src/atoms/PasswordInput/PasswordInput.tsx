import { TextInputProps } from '../TextInput/TextInput.types';
import { TextInputStyled } from '../TextInput/TextInput.styled';

export const PasswordInput = ({
    extraClassnames,
    ...props
}: TextInputProps) => (
    <TextInputStyled {...props} className={extraClassnames} type="password" />
);
