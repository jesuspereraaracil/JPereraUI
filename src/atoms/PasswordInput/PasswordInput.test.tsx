import { render } from '@testing-library/react';
import { PasswordInput } from './PasswordInput';

describe('PasswordInput', () => {
    test('Should render correctly', () => {
        const { container } = render(<PasswordInput error={true} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
