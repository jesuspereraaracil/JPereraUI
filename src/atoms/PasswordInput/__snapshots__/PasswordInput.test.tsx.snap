// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput Should render correctly 1`] = `
<input
  class="sc-bgqQcB irYhjw"
  type="password"
/>
`;
